### PR TITLE
plugin: initialize factor weights on plugin load

### PR DESCRIPTION
**Pull request by cmoussa1**
#### Problem

Mentioned as possible cleanup in #455, the weight of each factor is checked every time the priority of a job is calculated. It would be cleaner to put this check somewhere else so it does not need to be repeated for every job.

---

This PR creates defaults for the weights of each factor used in the priority calculation of a job and initializes each weight when the plugin is first loaded. In the callback for `conf.update`, it then checks to see if these weights have been configured; if not, it will just continue to use the defaults defined at the top of the file.

### CODE REVIEW

File: src/plugins/mf_priority.cpp [hunk 1]
```diff
@@ -201,8 +201,8 @@ static int conf_update_cb (flux_plugin_t *p,
     }
 
     // assign unpacked weights into priority_weights map
-    priority_weights["fairshare"] = fshare_weight;
-    priority_weights["queue"] = queue_weight;
+    if (fshare_weight != -1) priority_weights["fairshare"] = fshare_weight;
+    if (queue_weight != -1) priority_weights["queue"] = queue_weight;
```
Comment (grondo) [inline src/plugins/mf_priority.cpp:205]: I find single line conditionals a little hard to read, I'd suggest breaking these into two lines for readability.
I see we didn't comment on this earlier so this is just a suggestion not required.

**Review by grondo — APPROVED** [review]
This is great! Just one very minor comment.

File: src/plugins/mf_priority.cpp (hunk 1 above)
Comment (cmoussa1) [reply src/plugins/mf_priority.cpp]: No problem - I think that's perfectly reasonable. I'll push up a fix to break these up into two lines. 👍

**Comment by cmoussa1** [comment]
Thanks for reviewing @grondo! I'll set MWP here

**Pull request merged**